WPF/SeeShells/SeeShells/ShellParser/OfflineRegistryReader.cs

```diff
@@ -1,15 +1,139 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
 using System.Text;
-
+using Registry;
+using Registry.Abstractions;
 
 namespace SeeShells.ShellParser
 {
-    class OfflineRegistryReader : IRegistryReader
+    /// <summary>
+    /// This class is used to read an offline regisry hive file.
+    /// Using the Registry API, from EricZimmerman, it reads registry values from an hive file. 
+    /// </summary>
+    public class OfflineRegistryReader : IRegistryReader
     {
+        IConfigParser Parser { get; }
+
+        public OfflineRegistryReader(IConfigParser parser)
```

Comment:
As mentioned below, something should be passed into OfflineRegistryReader so that it doesnt have to obtain the file by itself, that way its not the Reader's job to handle file related management other than reading the file.
```suggestion
        public OfflineRegistryReader(IConfigParser parser, File registryFile)
```
```suggestion
        public OfflineRegistryReader(IConfigParser parser, String registryFile)
```